OEF-Social_Service/OEF-Social_Service.DataAccess/Data/Services/FollowService.cs

```diff
@@ -3,36 +3,160 @@
+                statementText.Append("MATCH (p1:Person), (p2:Person) WHERE p1.Id = $userId AND p2.Id = $userId2 CREATE (p1)-[p:Request_Send] ->(p2)");
+                var statementParameters = new Dictionary<string, object>
+                        {
+                        {"userId", personString},
+                        {"userId2", personString2}
+                        };
+                using (_session)
+                {
+                    //_session.LastBookmark.Values(null);
+                    var excecuteResult = await _session.RunAsync(statementText.ToString(), statementParameters);
+                }
+            }
+            if (relationExist == true)
+            {
+                return;
+            }
+        }
+
+        public async Task<string> GetRequests(Guid person)
+        {
+            var personString = person.ToString();
+            var data = new List<Person>();
+            var statementText = new StringBuilder();
+            statementText.Append("Match (user:Person)-[r:Request_Send]->({Id: $firstname}) Return user");
+            var statementParameters = new Dictionary<string, object>
+            {
+                {"firstname", personString},
+            };
+
+            using (_session)
+            {
+                var query = await _session.RunAsync(statementText.ToString(), statementParameters);
+                var result = await query.ToListAsync();
+                var serializedResult = JsonSerializer.Serialize(result);
+                return serializedResult;
+            }
+        }
+        public async Task<bool> DoesRelationExist(string person1, string person2)
+        {
+            var statementText = new StringBuilder();
+            statementText.Append("Match (p:Person {Id:$userId}), (b:Person {Id: $userId2}) RETURN EXISTS((p)-[]->(b))");
+            var statementParameters = new Dictionary<string, object>
+            {
+                {"userId", person1},
+                {"userId2", person2}
+            };
+
+            using (replicaSession)
+            {
+                var query = await replicaSession.RunAsync(statementText.ToString(), statementParameters);
+                var result = query.ToListAsync();
+                foreach (var item in result.Result[0].Values)
+                {
+                    if ((bool)item.Value == true)
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+        }
+
+        public async Task DeleteRelation(Guid person1, Guid person2)
+        {
+            var personString = person1.ToString();
+            var personString2 = person2.ToString();
+
+            var statementText = new StringBuilder();
+            statementText.Append("Match (user:Person)-[r]->({Id: $userId2}) Where ({Id: $userId})-[r]->() DELETE r");
+            var statementParameters = new Dictionary<string, object>
```

Comment:
Instead of Dictionaries you can use an OGM so we better follow SOLID principles. 
https://neo4j.com/developer/neo4j-ogm/

Reply:
OGM is indeed better but for this semester we keep these queries. and this can be done in a later semester (Refactor)